cmd/cogito/main_test.go

```diff
@@ -6,59 +6,66 @@ import (
 	"strings"
 	"testing"
 
+	"github.com/Pix4D/cogito/testhelp"
 	"gotest.tools/v3/assert"
 )
 
-func TestRunSmokeSuccess(t *testing.T) {
-	type testCase struct {
-		name string
-		args []string
-		in   string
-	}
-
-	test := func(t *testing.T, tc testCase) {
-		in := strings.NewReader(tc.in)
-		var out bytes.Buffer
-		var logOut bytes.Buffer
+func TestRunCheckSmokeSuccess(t *testing.T) {
+	in := strings.NewReader(`
+{
+  "source": {
+    "owner": "the-owner",
+    "repo": "the-repo",
+    "access_token": "the-secret",
+    "log_level": "debug"
+  }
+}`)
+	var out bytes.Buffer
+	var logOut bytes.Buffer
 
-		err := run(in, &out, &logOut, tc.args)
+	err := run(in, &out, &logOut, []string{"check"})
 
-		assert.NilError(t, err, "\nout: %s\nlogOut: %s", out.String(), logOut.String())
-	}
+	assert.NilError(t, err, "\nout: %s\nlogOut: %s", out.String(), logOut.String())
+}
 
-	testCases := []testCase{
-		{
-			name: "check",
-			args: []string{"check"},
-			in: `
+func TestRunGetSmokeSuccess(t *testing.T) {
+	in := strings.NewReader(`
 {
   "source": {
     "owner": "the-owner",
     "repo": "the-repo",
-    "access_token": "the-secret"
-  }
-}`,
-		},
-		{
-			name: "get",
-			args: []string{"in", "dummy-dir"},
-			in: `
+    "access_token": "the-secret",
+    "log_level": "debug"
+  },
+  "version": {"ref": "pizza"}
+}`)
+	var out bytes.Buffer
+	var logOut bytes.Buffer
+
+	err := run(in, &out, &logOut, []string{"in", "dummy-dir"})
+
+	assert.NilError(t, err, "\nout: %s\nlogOut: %s", out.String(), logOut.String())
+}
+
+func TestRunPutSmokeSuccess(t *testing.T) {
+	in := strings.NewReader(`
 {
   "source": {
     "owner": "the-owner",
     "repo": "the-repo",
-    "access_token": "the-secret"
+    "access_token": "the-secret",
+    "log_level": "debug"
   },
-  "version": {"ref": "pizza"}
-}`,
-		},
-	}
+  "params": {"state": "pending"}
+}`)
+	var out bytes.Buffer
+	var logOut bytes.Buffer
+	inputDir := testhelp.MakeGitRepoFromTestdata(t, "../../cogito/testdata/one-repo/a-repo",
+		testhelp.HttpsRemote("the-owner", "the-repo"), "dummySHA", "dummyHead")
 
-	for _, tc := range testCases {
-		t.Run(tc.name, func(t *testing.T) {
-			test(t, tc)
-		})
-	}
+	err := run(in, &out, &logOut, []string{"out", inputDir})
```

Comment:
I also transformed some tests to the new pattern with the `test()` anonymous function that I already introduced in this PR series; it makes the test _intent_ more evident because now it appears before the (long) list of test cases.